submissions/chris-voitova/friends-app/script.js

```diff
@@ -0,0 +1,199 @@
+      sortFunc(filteredUsers);
+      renderUsers(filteredUsers);
+    }
+    if (target === descendingTrigger) {
+      sortFunc(filteredUsers).reverse();
+      renderUsers(filteredUsers);
+    }
+  });
+};
+
+const checkSelectedState = (users) => {
+  let data;
+  state.maleGender
+    ? (data = filterByGender(users, "male"))
+    : state.femaleGender
+    ? (data = filterByGender(users, "female"))
+    : (data = users);
+  return data;
+};
+
+const handleFilterByGender = (users, initUsers) => {
+  const maleTrigger = document.querySelector("#gender-male");
+  const femaleTrigger = document.querySelector("#gender-female");
+  const allGendersTrigger = document.querySelector("#gender-all");
+  const genderBlock = document.querySelector("#gender-block");
+
+  genderBlock.addEventListener("click", ({ target }) => {
+    if (target === maleTrigger) {
+      resetState();
+      resetSorting();
+      const men = filterByGender(users, "male");
+      renderUsers(men);
+      state.maleGender = true;
+    }
+    if (target === femaleTrigger) {
+      resetState();
+      resetSorting();
+      const women = filterByGender(users, "female");
+      renderUsers(women);
+      state.femaleGender = true;
+    }
+    if (target === allGendersTrigger) {
+      resetState();
+      resetSorting();
+      renderUsers(initUsers);
+      state.allGenders = true;
+    }
+  });
+};
+
+const resetState = () => {
+  for (let stateItem in state) {
+    if (state.hasOwnProperty(stateItem)) {
+      state[stateItem] = false;
+    }
+  }
+};
+
+const resetSorting = () => {
+  const sortingFields = document.querySelectorAll(
+    "#sort-ascending, #sort-descending, #sort-az, #sort-za"
+  );
+  [...sortingFields].forEach((field) => (field.checked = false));
+};
+
+const handleResetToDefault = (initUsers) => {
+  const resetBtn = document.querySelector("#reset-btn");
+  resetBtn.addEventListener("click", () => {
+    renderUsers(initUsers);
+    state = initState;
+  });
+};
+const handleSearch = (users) => {
+  const searchInput = document.querySelector("#search-input");
+  searchInput.addEventListener("input", ({ target: { value } }) => {
+    searchUsers(users, value);
+  });
+};
+
+const sortByAge = (users) => {
+  return users.sort((a, b) => a.dob.age - b.dob.age);
+};
+const sortByName = (users) => {
+  return users.sort((a, b) => (a.name.first > b.name.first ? 1 : -1));
+};
+const filterByGender = (users, gender) => {
+  return users.filter(({ gender: userGender }) => userGender === gender);
+};
+
+const searchUsers = (users, searchValue) => {
+  searchValue = searchValue.toLowerCase();
+  if (searchValue) {
+    const searchResults = users.filter(({ name: { first: firstName } }) => {
+      firstName = firstName.toLowerCase();
+      return firstName.includes(searchValue);
+    });
+    renderUsers(searchResults);
+
+    if (searchResults.length === 0) {
+      usersContainer.innerHTML = "";
+      usersContainer.innerHTML += errorMessage("users not found :(");
```

Comment:
Why it is error? Error is when something is broken.

Empty results is a normal situation :)